Formula/python.rb

```diff
@@ -68,9 +68,9 @@ class Python < Formula
     sha256 "c075353337f9ff3ccf8091693d278782fcdff62c113245d8de43c5c7acc57daf"
   end
 
-  # Patch for compiling universal binaries on macOS 10.12
+  # Patch for compiling universal binaries on macOS 10.12, as well on 10.11 with Xcode 8
   # https://bugs.python.org/issue27806
-  if MacOS.version >= :sierra && build.universal?
+  if DevelopmentTools.clang_build_version >= 800 && build.universal?
```

Comment:
You should reverse the order of these two conditions since the second one is less common than the first.